NinchatSDKSwift/Implementations/Data Source - Delegate/NINWebRTCDelegate.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  NINWebRTCDelegate.swift
+//  NinchatSDKSwift
+//
+//  Created by Hassan Shahbazi on 25.12.2019.
+//  Copyright © 2019 Hassan Shahbazi. All rights reserved.
+//
+
+import Foundation
+import NinchatSDK
+
+protocol NINWebRTCActions {
+    var onError: ((Error) -> Void)? { get set }
+    var onRemoteVideoTrack: ((RTCVideoRenderer, RTCVideoTrack) -> Void)? { get set }
+    var onLocalCapturer: ((RTCCameraVideoCapturer) -> Void)? { get set }
+}
+
+protocol NINWebRTCDelegate: NINWebRTCClientDelegate, NINWebRTCActions {
+    init(remoteVideoDelegate: RTCVideoViewDelegate)
+}
+
+final class NINWebRTCDelegateImpl: NSObject, NINWebRTCDelegate {
+    
+    // MARK: - NINWebRTCActions
+    
+    var onError: ((Error) -> Void)?
+    var onRemoteVideoTrack: ((RTCVideoRenderer, RTCVideoTrack) -> Void)?
+    var onLocalCapturer: ((RTCCameraVideoCapturer) -> Void)?
+    
+    // MARK: - NINWebRTCDelegate
+    
+    private let remoteVideoDelegate: RTCVideoViewDelegate
+    
+    init(remoteVideoDelegate: RTCVideoViewDelegate) {
+        self.remoteVideoDelegate = remoteVideoDelegate
+    }
+}
+
+// MARK: - NINWebRTCClientDelegate
+
+extension NINWebRTCDelegateImpl {
+    func webrtcClient(_ client: NINWebRTCClient!, didGetError error: Error!) {
+        #if DEBUG
+        print("NINCHAT: didGetError: \(String(describing: error))")
```

Comment:
How about creating a debug wrapper for the print function, and making it a nop #if DEBUG?

Reply:
Very good point. Thanks